src/main/java/com/upserve/uppend/lookup/LookupMetadata.java

```diff
@@ -156,21 +203,36 @@ public Long findKey(VirtualLongBlobStore longBlobStore, LookupKey key) {
             if (comparison < 0) {
```

Comment:
Since this is the positive comparison, I'd put this one first:
```
if(comparison == 0) {
  key.setPosition(keyPosition);
  hitCount.increment();
  return longBlobStore.readLong(keyPosition);
}
```
The remaining if/else can be a bit simpler:
```
if (comparison < 0) {
  upperKey = midpointKey;
  keyIndexUpper = midpointKeyIndex;
  bisectKeyTreeArrayIndex = bisectKeyTreeArrayIndex * 2;
} else
  lowerKey = midpointKey;
  keyIndexLower = midpointKeyIndex;
  bisectKeyTreeArrayIndex = bisectKeyTreeArrayIndex * 2 + 1;
}
```